edk2toolext/edk2_logging.py

```diff
@@ -201,20 +201,28 @@ def scan_compiler_output(output_stream):
     # seek to the start of the output stream
     def output_compiler_error(match, line, start_txt="Compiler"):
         start, end = match.span()
-        source = line[:start]
-        error = line[end:]
+        source = line[:start].strip()
+        error = line[end:].strip()
         num = match.group(1)
         return f"{start_txt} #{num} from {source} {error}"
     problems = []
     output_stream.seek(0, 0)
     error_exp = re.compile(r"error [A-EG-Z]?(\d+):")
+    # Would prefer to do something like r"(?:\/[^\/: ]*)+\/?:\d+:\d+: (error):"
+    # but the script is currently setup on fixed formatting assumptions rather
+    # than offering per rule flexibility to parse tokens via regex
+    gcc_error_exp = re.compile(r":\d+:\d+: (error):")
     edk2_error_exp = re.compile(r"error F(\d+):")
     build_py_error_exp = re.compile(r"error (\d+)E:")
     linker_error_exp = re.compile(r"error LNK(\d+):")
     warning_exp = re.compile(r"warning [A-Z]?(\d+):")
     for raw_line in output_stream.readlines():
         line = raw_line.strip("\n").strip()
         match = error_exp.search(line)
+        if match is not None:
```

Comment:
Any thoughts on adding a break after we match any of the below if statements? I presume once we find one, we can move to the next line instead of checking the same line for all other error types.

Reply:
I'd need a bit more time to confirm that. Because the regex matches fragments of an overall line string it is possible (though I think unlikely) that more than one match occurs on a line.

I would prefer to hold off on that to prevent potentially breaking an obscure case until more investigation is done.